### PR TITLE
fix: type error in AuthScreen.tsx

### LINKED ISSUE #457: Fix type errors in AuthScreen.tsx
### Issue Description
In [AuthScreen.tsx](https://github.com/Real-Dev-Squad/mobile-app/blob/develop/src/screens/AuthScreen/AuthScreen.tsx) file, there are some issues like data types of variables are not explicitly defined, and unnecessary comments are added. 

### Expected Behavior
This [AuthScreen.tsx](https://github.com/Real-Dev-Squad/mobile-app/blob/develop/src/screens/AuthScreen/AuthScreen.tsx) file should be well structured. There should not be any errors or unnecessary comments.

### Screenshots

image
image

### Environment

- Operating System: [e.g., Windows, macOS, Linux]
- Browser (if applicable): [e.g., Chrome, Firefox, Safari]
- Version of Software/Project: [e.g., v1.0.0]

### Checklist

- [x] I have read and followed the project's code of conduct.
- [x] I have searched for similar issues before creating this one.
- [x] I have provided all the necessary information to understand and reproduce the issue.
- [x] I am willing to contribute to the resolution of this issue.

---

Thank you for contributing to our project! We appreciate your help in improving it.

📚 See [contributing instructions](https://github.com/samyakshah3008/mobile-app/blob/develop/CONTRIBUTING.md).

### PR DESCRIPTION
**Pull request by surajmaity1**
Date: 09-10-2024

Developer Name: Suraj Maity

---

## Issue Ticket Number

- Closes #457 
## Description

- Variables' data type explicitly defined.
- Unnecessary comments removed.

### Documentation Updated?

- [ ] Yes
- [x] No

### Under Feature Flag

- [ ] Yes
- [x] No

### Database Changes

- [ ] Yes
- [x] No

### Breaking Changes

- [ ] Yes
- [x] No

### Development Tested?

- [x] Yes
- [ ] No

## Screenshots
<details>
<summary>Screenshot 1</summary>

https://github.com/user-attachments/assets/520381dd-fef3-4983-a111-42698f03a851

</details>

## Test Coverage
<details>
<summary>Screenshot 1</summary>

image

</details>

### CODE REVIEW

File: src/screens/AuthScreen/AuthScreen.tsx [hunk 1]
```diff
@@ -274,20 +273,6 @@ const AuthScreen = () => {
...
-              ? dispatch({ type: 'DEV' })
-              : dispatch({ type: 'PROD' });
-          }}
-        >
-          <View style={AuthViewStyle.signInTxtView}>
-            <Text style={AuthViewStyle.signInText}>
-              {isProdEnvironment ? 'Switch to DEV' : 'Switch to Prod'}
-            </Text>
-          </View>
-        </TouchableOpacity> */}
```
Comment (Ayushsanjdev) [inline src/screens/AuthScreen/AuthScreen.tsx:290]: Might need to confirm with the developer who wrote it, is it required or not.

**Comment by surajmaity1** [comment]
@ashifkhn please review and approve this PR.

File: src/context/type.ts [hunk 2]
```diff
@@ -15,3 +20,9 @@ export interface User {
   twitterUrl: string;
   githubUrl: string;
 }
+
+export interface RootState {
```
Comment (vinit717) [inline src/context/type.ts]: Why it is interface and some other place it is type

**Review by shreya-mishra — DISMISSED** [review]

**Comment by vinit717** [comment]
Can you add the screenshot

**Comment by surajmaity1** [comment]
> Can you add the screenshot

Screenshot 2024-10-19 194054

Screenshot 2024-10-19 194133

**Comment by surajmaity1** [comment]
> Can you add the screenshot

Screenshots added.

**Review by Ayushsanjdev — CHANGES_REQUESTED** [review]

File: src/screens/AuthScreen/AuthScreen.tsx [hunk 3]
```diff
@@ -274,30 +278,16 @@ const AuthScreen = () => {
...
-        <CameraScreen
+        <Camera
           style={StyleSheet.absoluteFill}
           showFrame={true}
           scanBarcode={true}
           onReadCode={handleQRCodeScanned}
           frameColor={'white'}
           laserColor={'white'}
-          hideControls
+          hideControls={true}
```
Comment (Ayushsanjdev) [inline src/screens/AuthScreen/AuthScreen.tsx]: We don't need true explicitly

**Review by Ayushsanjdev — CHANGES_REQUESTED** [review]

File: src/react-native-walkthrough-tooltip.d.ts [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    content?: React.ReactElement;
+    placement?: 'top' | 'bottom' | 'left' | 'right' | 'center';
+    onClose?: (event: GestureResponderEvent) => void;
+    closeOnContentInteraction?: boolean;
+    closeOnBackgroundInteraction?: boolean;
+    backgroundStyle?: ViewStyle;
+    contentStyle?: ViewStyle;
+    tooltipStyle?: ViewStyle;
+    arrowStyle?: ViewStyle;
+    children?: React.ReactNode;
```
Comment (Ayushsanjdev) [inline src/react-native-walkthrough-tooltip.d.ts:16]: Everything optional?

File: src/screens/AuthScreen/AuthScreen.tsx [hunk 5]
```diff
@@ -220,7 +223,8 @@ const AuthScreen = () => {
       console.log(err, 'Error QR login');
       Toast.show({
         type: 'error',
-        text1: err,
+        text1:
```
Comment (Ayushsanjdev) [inline src/screens/AuthScreen/AuthScreen.tsx:223]: Why name is text1?

File: src/screens/AuthScreen/AuthScreen.tsx [hunk 6]
```diff
@@ -274,30 +278,16 @@ const AuthScreen = () => {
...
       </View>
       {cameraActive && (
-        <CameraScreen
+        <Camera
           style={StyleSheet.absoluteFill}
           showFrame={true}
           scanBarcode={true}
           onReadCode={handleQRCodeScanned}
           frameColor={'white'}
           laserColor={'white'}
```
Comment (Ayushsanjdev) [inline src/screens/AuthScreen/AuthScreen.tsx:286]: No explicit true + can use only strings without brackets

**Review by Ayushsanjdev — CHANGES_REQUESTED** [review]

File: src/screens/AuthScreen/AuthScreen.tsx (hunk 1 above)
Comment (surajmaity1) [reply src/screens/AuthScreen/AuthScreen.tsx:290]: It was not used. That's why I removed.

File: src/react-native-walkthrough-tooltip.d.ts (hunk 4 above)
Comment (surajmaity1) [reply src/react-native-walkthrough-tooltip.d.ts:16]: Yes. If you check react-native-walkthrough-tooltip module, you'll find it.

File: src/screens/AuthScreen/AuthScreen.tsx (hunk 3 above)
Comment (surajmaity1) [reply src/screens/AuthScreen/AuthScreen.tsx]: It is modified.

File: src/screens/AuthScreen/AuthScreen.tsx (hunk 1 above)
Comment (Ayushsanjdev) [reply src/screens/AuthScreen/AuthScreen.tsx:290]: okay

File: src/screens/AuthScreen/AuthScreen.tsx (hunk 5 above)
Comment (Ayushsanjdev) [reply src/screens/AuthScreen/AuthScreen.tsx:223]: please let me know this

**Review by Ayushsanjdev — APPROVED** [review]

File: src/screens/AuthScreen/AuthScreen.tsx (hunk 6 above)
Comment (Ayushsanjdev) [reply src/screens/AuthScreen/AuthScreen.tsx:286]: not resolved yet

File: src/react-native-walkthrough-tooltip.d.ts [hunk 7]
```diff
@@ -0,0 +1,20 @@
+declare module 'react-native-walkthrough-tooltip' {
+  import React from 'react';
+  import { GestureResponderEvent, ViewStyle } from 'react-native';
+
+  export interface TooltipProps {
```
Comment (vinit717) [inline src/react-native-walkthrough-tooltip.d.ts:5]: Why interface here as loggedInUserType is using type

File: src/screens/AuthScreen/AuthScreen.tsx [hunk 8]
```diff
@@ -220,7 +223,8 @@ const AuthScreen = () => {
       console.log(err, 'Error QR login');
```
Comment (vinit717) [inline src/screens/AuthScreen/AuthScreen.tsx]: Why the log statement is there?

File: src/screens/AuthScreen/AuthScreen.tsx [hunk 9]
```diff
@@ -19,20 +19,25 @@ import { AuthContext } from '../../context/AuthContext';
 import { getUserData, goalsAuth, requestCameraPermission } from './Util';
 import { storeData } from '../../utils/dataStore';
 import AuthApis from '../../constants/apiConstant/AuthApi';
-// import { AuthApisStaging } from '../../constants/apiConstant/AuthApi';
-import { CameraScreen } from 'react-native-camera-kit';
+import { Camera } from 'react-native-camera-kit';
```
Comment (vinit717) [inline src/screens/AuthScreen/AuthScreen.tsx:22]: Why is this changed to the camera?

**Review by vinit717 — CHANGES_REQUESTED** [review]

File: src/screens/AuthScreen/AuthScreen.tsx (hunk 9 above)
Comment (surajmaity1) [reply src/screens/AuthScreen/AuthScreen.tsx:22]: You can check here - 
https://github.com/teslamotors/react-native-camera-kit?tab=readme-ov-file#barcode--qr-code-scanning

Please check that CameraScreen was removed - https://github.com/teslamotors/react-native-camera-kit/commit/5a7df3ee804c189570099426b43cc9676ee6b0b5

Comment (surajmaity1) [reply src/screens/AuthScreen/AuthScreen.tsx:22]: Also check screenshots:
Screenshot 2024-10-21 063907

Screenshot 2024-10-21 064130

File: src/screens/AuthScreen/AuthScreen.tsx (hunk 8 above)
Comment (surajmaity1) [reply src/screens/AuthScreen/AuthScreen.tsx]: It is fixed.

File: src/react-native-walkthrough-tooltip.d.ts (hunk 7 above)
Comment (surajmaity1) [reply src/react-native-walkthrough-tooltip.d.ts:5]: loggedInUserType was not created by me. I'll create a separate PR for it.

**Review by iamitprakash — DISMISSED** [review]
why we need to store token and why we need social media i'ds

**Comment by surajmaity1** [comment]
> why we need to store token and why we need social media i'ds

Here, token is already being used multiple purpose. Here my task is to fix type error in AuthScreen.tsx.

**Comment by surajmaity1** [comment]
I've changed CameraScreen to Camera as it deprecates.

**Review by vinit717 — APPROVED** [review]

**Pull request merged**